src/main/java/io/openliberty/tools/intellij/util/Constants.java

```diff
@@ -50,16 +51,33 @@
     public static final String VIEW_GRADLE_CONFIG_ACTION_ID = "io.openliberty.tools.intellij.actions.ViewGradleConfig";
     public static final String VIEW_EFFECTIVE_POM_ACTION_ID = "io.openliberty.tools.intellij.actions.ViewEffectivePom";
 
-    public static final Map<String, String> ACTIONS_MAP = new HashMap<String, String>() {
+    public static HashMap<String, String> CORE_ACTIONS_MAP = new HashMap<String, String>() {
         {
             put(LIBERTY_DEV_START, LIBERTY_DEV_START_ACTION_ID);
             put(LIBERTY_DEV_STOP, LIBERTY_DEV_STOP_ACTION_ID);
             put(LIBERTY_DEV_CUSTOM_START, LIBERTY_DEV_CUSTOM_START_ACTION_ID);
             put(LIBERTY_DEV_TESTS, LIBERTY_DEV_TESTS_ACTION_ID);
-            put(VIEW_UNIT_TEST_REPORT, VIEW_UNIT_TEST_REPORT_ACTION_ID);
-            put(VIEW_INTEGRATION_TEST_REPORT, VIEW_INTEGRATION_TEST_REPORT_ACTION_ID);
-            put(VIEW_GRADLE_TEST_REPORT, VIEW_GRADLE_TEST_REPORT_ACTION_ID);
         }
     };
 
+    public static HashMap<String, String> getFullActionMap() {
+        HashMap<String, String> FULL_ACTIONS_MAP = CORE_ACTIONS_MAP;
+        FULL_ACTIONS_MAP.put(VIEW_UNIT_TEST_REPORT, VIEW_UNIT_TEST_REPORT_ACTION_ID);
+        FULL_ACTIONS_MAP.put(VIEW_INTEGRATION_TEST_REPORT, VIEW_INTEGRATION_TEST_REPORT_ACTION_ID);
+        FULL_ACTIONS_MAP.put(VIEW_GRADLE_TEST_REPORT, VIEW_GRADLE_TEST_REPORT_ACTION_ID);
+        return FULL_ACTIONS_MAP;
+    }
+    
+    public static HashMap<String, String> getMavenMap() {
+        HashMap<String, String> MAVEN_ACTIONS_MAP = CORE_ACTIONS_MAP;
+        MAVEN_ACTIONS_MAP.put(VIEW_UNIT_TEST_REPORT, VIEW_UNIT_TEST_REPORT_ACTION_ID);
+        MAVEN_ACTIONS_MAP.put(VIEW_INTEGRATION_TEST_REPORT, VIEW_INTEGRATION_TEST_REPORT_ACTION_ID);
+        return MAVEN_ACTIONS_MAP;
+    }
+
+    public static HashMap<String, String> getGradleMap() {
+        HashMap<String, String> GRADLE_ACTIONS_MAP = CORE_ACTIONS_MAP;
+        GRADLE_ACTIONS_MAP.put(VIEW_GRADLE_TEST_REPORT, VIEW_GRADLE_TEST_REPORT_ACTION_ID);
+        return GRADLE_ACTIONS_MAP;
+    }
```

Comment:
Note that doing like `HashMap<String, String> a = b;` does not make a copy of the map.  So when you are doing `put` afterwards in these 3 methods, you are actually modifying the original map.  To make a copy, you could do `a = new HashMap<String, String>()` then `a.putAll(b)`

Reply:
Also things like FULL_ACTIONS_MAP in the method is a local variable and not a constant, so convention should be ordinary camel case.